harvester/harvester
# Get required image info for UpgradeLog from cluster repo

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**


We hardcoded the UpgradeLog-related image repos and tags in the controller code base.

**Solution:**


Determine the image repos and tags by querying the corresponding charts from the cluster repo container at runtime.

**Related Issue:**

#3595 

**Test plan:**


1. Build a new ISO image containing this PR and provision a single node cluster with it
2. Upgrade the cluster with the same ISO image
3. Check the `UpgradeLog` CR to see if `InfraReady` condition is `True`
4. After the `DownloadReady` becomes `True`, try to download the log archive
